torch/_inductor/simple_fsdp/reorder.py

```diff
@@ -34,47 +33,25 @@ def reorder_all_gather(
             # gather i-th all gather node and its dependencies
             all_gather_list.append(node)
             inverse_user = list(inverse_users[node])
-            if (
-                len(inverse_user) > 0
-                and not node_to_type[inverse_user[0]] == NodeType.ALL_GATHER
-                and inverse_user not in all_gather_list
-            ):
+            if len(inverse_user) > 0:
                 all_gather_list.extend(inverse_user)
         elif node_type == NodeType.AG_WAIT:
             if (
-                (
-                    (
-                        node_to_type[snodes[idx + 1]] == NodeType.AG_WAIT
-                        and node_to_type[snodes[idx + 2]] == NodeType.ALL_GATHER
-                    )
-                    or (
-                        node_to_type[snodes[idx + 1]] == NodeType.ALL_GATHER
-                        and node_to_type[snodes[idx - 1]] != NodeType.AG_WAIT
-                    )
-                )
-                and not all_gather_before_last_wait
+                not all_gather_before_last_wait
                 and len(all_gather_list) > 0
             ):
+                assert node_to_type[snodes[idx + 1]] is NodeType.ALL_GATHER
```

Comment:
nit: we should consistently use `is` vs. `==` for `NodeType` checking

Reply:
Thank you! Updated.